huaweicloud/services/cae/resource_huaweicloud_cae_compontent.go

```diff
@@ -0,0 +1,535 @@
+	}
+
+	return []map[string]interface{}{
+		{
+			"archive": []map[string]interface{}{
+				{
+					"artifact_namespace": utils.PathSearch("artifact_namespace", archive, nil),
+				},
+			},
+			"parameters": parameters,
+		},
+	}
+}
+
+func resourceComponentUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg         = meta.(*config.Config)
+		region      = cfg.GetRegion(d)
+		httpUrl     = "v1/{project_id}/cae/applications/{application_id}/components/{component_id}"
+		product     = "cae"
+		client, err = cfg.NewServiceClient(product, region)
+	)
+	if err != nil {
+		return diag.Errorf("error creating CAE Client: %s", err)
+	}
+
+	updatePath := client.Endpoint + httpUrl
+	updatePath = strings.ReplaceAll(updatePath, "{project_id}", client.ProjectID)
+	updatePath = strings.ReplaceAll(updatePath, "{application_id}", d.Get("application_id").(string))
+	updatePath = strings.ReplaceAll(updatePath, "{component_id}", d.Id())
+
+	updateComponentOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"X-Environment-Id": d.Get("environment_id").(string),
+		},
+		JSONBody: utils.RemoveNil(buildCreateOrUpdateComponentBodyParams(d)),
+	}
+	_, err = client.Request("PUT", updatePath, &updateComponentOpt)
+	if err != nil {
+		return diag.Errorf("error updating CAE component: %s", err)
+	}
+
+	return resourceComponentRead(ctx, d, meta)
+}
+
+func resourceComponentDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg         = meta.(*config.Config)
+		region      = cfg.GetRegion(d)
+		httpUrl     = "v1/{project_id}/cae/applications/{application_id}/components/{component_id}"
+		product     = "cae"
+		client, err = cfg.NewServiceClient(product, region)
+	)
+	if err != nil {
+		return diag.Errorf("error creating CAE Client: %s", err)
+	}
+
+	deletePath := client.Endpoint + httpUrl
+	deletePath = strings.ReplaceAll(deletePath, "{project_id}", client.ProjectID)
+	deletePath = strings.ReplaceAll(deletePath, "{application_id}", d.Get("application_id").(string))
+	deletePath = strings.ReplaceAll(deletePath, "{component_id}", d.Id())
+
+	deleteComponentOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"X-Environment-Id": d.Get("environment_id").(string),
+		},
+	}
+	_, err = client.Request("DELETE", deletePath, &deleteComponentOpt)
+	if err != nil {
+		return diag.Errorf("error deleting CAE component: %s", err)
+	}
+
+	return nil
+}
+
+func resourceComponentImportState(_ context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData, error) {
+	importedId := d.Id()
+	parts := strings.Split(importedId, "/")
+	if len(parts) != 3 {
+		return nil, fmt.Errorf("invalid format specified for import ID, want '<environment_id>/<application_id>/<id>', but got '%s'",
+			importedId)
+	}
+
+	d.SetId(parts[2])
+	mErr := multierror.Append(nil,
+		d.Set("application_id", parts[1]),
```

Comment:
did we forgot the environment ID setting?

Reply:
fixed